ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OAutoML.scala

```diff
@@ -124,17 +127,28 @@ class H2OAutoML(override val uid: String) extends Estimator[H2OMOJOModel]
     getIncludeAlgos().diff(bothIncludedExcluded).map(Algo.valueOf)
   }
 
-  private def leaderboardAsSparkFrame(aml: AutoML): Option[DataFrame] = {
+  private def leaderboardAsSparkFrame(aml: AutoML, extraColumns: Array[String]): DataFrame = {
     // Get LeaderBoard
-    val twoDimtable = aml.leaderboard().toTwoDimTable()
-    val colNames = twoDimtable.getColHeaders
-    val data = aml.leaderboard().toTwoDimTable().getCellValues.map(_.map(_.toString))
+    val twoDimTable = aml.leaderboard().toTwoDimTable(extraColumns: _*)
+    val colNames = twoDimTable.getColHeaders
+    val data = twoDimTable.getCellValues.map(_.map(_.toString))
     val rows = data.map {
       Row.fromSeq(_)
     }
     val schema = StructType(colNames.map { name => StructField(name, StringType) })
     val rdd = spark.sparkContext.parallelize(rows)
-    Some(spark.createDataFrame(rdd, schema))
+    spark.createDataFrame(rdd, schema)
+  }
+
+  def getLeaderboard(extraColumns: String*): DataFrame = getLeaderboard(extraColumns.toArray)
+
+  def getLeaderboard(extraColumns: java.util.ArrayList[String]) : DataFrame = {
+    getLeaderboard(extraColumns.asScala.toArray)
+  }
+
+  def getLeaderboard(extraColumns: Array[String]): DataFrame = amlOption match {
+    case Some(aml) => leaderboardAsSparkFrame(aml, extraColumns)
+    case None => throw new RuntimeException("The 'fit' method must be called at first!")
```

Comment:
are `String*` and `Array[String]` really different? I thought the compiler would automatically translate the first into the second: even Java does that.

Reply:
@sebhrusen variable parameters in Scala are represented by seqence. If you want to pass collection to variable parameters you must use a special operator for that. `getLeaderboard(collection: _*)`. The method overloading with `java.util.ArrayList[String]` must be there because of Py4J and Python client. 

The overloading with `Array[String]` is unnecessary, you are right!